app/controllers/posts_controller.rb

```diff
@@ -36,6 +36,28 @@ def create
     end
   end
 
+  def update_post_destroy_first
+    post_counter = author.posts_counter
+    post_counter -= 1
+    User.find(author_id).update(posts_counter: post_counter)
+  end
+
+  # Get users/:user_id/posts/:id
+  def destroy
+    @user = User.find(params[:user_id])
+    @post = Post.find(params[:id])
+
+    @post.comments.destroy_all
+    @post.likes.destroy_all
+
+    @post.destroy
+
```

Comment:
- [ ] Good job implementing the destroy actions in the PostsController and in the CommentsController as requested, however a user is currently unable to create a post and comment, therefore I am unable to check the functionality of the app. Please remember that:

- A user can delete a post if it is theirs or if they have an admin role (column role has value "admin"). Use CanCanCan for this authorization.
For that, you need to implement the `post deleting` functionality. Add the "Delete" button to the view and make sure that only authorized users can see it.
- A user can delete a comment if it is theirs or if they have an admin role (column role has value "admin"). Use CanCanCan for this authorization.
For that, you need to implement the `comment deleting` functionality. Add the "Delete" button to the view and make sure that only authorized users can see it.



VideoCast
[cinnamon-2024-01-26T190439+0200.webm](https://github.com/simplegoose/blog_app/assets/81932478/a2978ec9-0094-423f-a678-2b81820d5583)